File: src/main/java/info/skyblond/archivedag/model/entity/GroupUserEntity.java
package info.skyblond.archivedag.model.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "group_user")
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@IdClass(GroupUserEntity.class)
public class GroupUserEntity implements Serializable {
    @Id
    @Column(name = "group_name", updatable = false)
    String groupName;

    @Id
    @Column(name = "username", nullable = false)
    String username;

    public GroupUserEntity(String groupName, String username) {
        this.groupName = groupName;
        this.username = username;
    }
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arstue/entity/GroupUserEntity.kt
package info.skyblond.archivedag.arstue.entity

import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "group_user")
@IdClass(GroupUserEntity::class)
class GroupUserEntity(
    @Id
    @Column(name = "group_name", updatable = false)
    val groupName: String,

    @Id
    @Column(name = "username", nullable = false)
    val username: String,
) : Serializable
